core/block.js

```diff
@@ -1630,111 +1630,201 @@ Blockly.Block.prototype.mixin = function(mixinObj, opt_disableCheck) {
 Blockly.Block.prototype.interpolate_ = function(message, args, lastDummyAlign,
     warningPrefix) {
   var tokens = Blockly.utils.tokenizeInterpolation(message);
-  // Interpolate the arguments.  Build a list of elements.
-  var indexDup = [];
-  var indexCount = 0;
-  var elements = [];
+  this.validateTokens_(tokens, args.length);
+  var elements = this.interpolateArguments_(tokens, args, lastDummyAlign);
+
+  // An array of [field, fieldName] tuples.
+  var fieldStack = [];
+  for (var i = 0, element; (element = elements[i]); i++) {
+    switch (element['type']) {
```

Comment:
I'm not sure either haha. My understanding is that since this is parsing JSON (which comes from outside the "compiled" blockly implementation) it's supposed to use the bracket notation. If it doesn't closure could rename this to something like `a.a` which wouldn't work with a `{ 'type': 'input_statement'}` JSON object. But I'm only an 80% on that =)

Reply:
I think you're right! Now that you mention it, I think I had the exact same conversation with Abby about toolbox json, so maybe I will remember this next time. Thanks!

Reply:
Hopefully I can remember next time too haha. After working on interpolate_ I realized I forgot to use the bracket notation in the type hierarchy D: Shouldn't be too hard to fix though hehe